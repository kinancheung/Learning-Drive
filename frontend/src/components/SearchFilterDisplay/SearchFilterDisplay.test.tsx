import { render, screen } from '@testing-library/react';

import { SearchFilterDisplay } from '.';

describe('SearchFilterDisplay', () => {
  it('renders without error', () => {
    const search = { searchQuery: 'Java' };

    const view = render(<SearchFilterDisplay search={search} clearSearch={jest.fn()} />);

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('renders blank with no search query', () => {
    const search = {};

    const view = render(<SearchFilterDisplay search={search} clearSearch={jest.fn()} />);

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('renders categories without error', () => {
    const search = { searchQuery: 'Java', searchCategories: ['Java'] };

    const view = render(<SearchFilterDisplay search={search} clearSearch={jest.fn()} />);

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('calls clearSearch when clear button pressed', () => {
    const search = { searchQuery: 'Java', searchCategories: ['Java'] };

    const clearFn = jest.fn();

    render(<SearchFilterDisplay search={search} clearSearch={clearFn} />);

    screen.getByTestId('clear-button').click();

    expect(clearFn).toHaveBeenCalled();
  });
});
