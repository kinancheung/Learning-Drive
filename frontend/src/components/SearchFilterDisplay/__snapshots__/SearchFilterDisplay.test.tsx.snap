// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilterDisplay renders blank with no search query 1`] = `<DocumentFragment />`;

exports[`SearchFilterDisplay renders categories without error 1`] = `
<DocumentFragment>
  <div
    class="css-mqlynq-MuiStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="filterIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.0929 2.57912C1.25675 2.22596 1.61069 2 2.00001 2H22C22.3893 2 22.7433 2.22596 22.9071 2.57912C23.071 2.93229 23.015 3.34845 22.7636 3.64573L15 12.8261V21C15 21.3466 14.8206 21.6684 14.5257 21.8507C14.2309 22.0329 13.8628 22.0494 13.5528 21.8944L9.5528 19.8944C9.21402 19.725 9.00001 19.3788 9.00001 19V12.8261L1.23644 3.64573C0.985046 3.34845 0.929037 2.93229 1.0929 2.57912ZM4.15532 4L10.7636 11.8143C10.9162 11.9948 11 12.2236 11 12.46V18.382L13 19.382V12.46C13 12.2236 13.0838 11.9948 13.2364 11.8143L19.8447 4H4.15532Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </svg>
    <div
      class="css-1d0f7f9-MuiStack-root"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 css-14wykv5-MuiTypography-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-1vsculk-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            Java
          </span>
        </div>
        Posts containing "Java"
      </div>
    </div>
    <div
      class="MuiBox-root css-i9gxme"
    />
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-qj1k01-MuiButtonBase-root-MuiButton-root"
      data-testid="clear-button"
      tabindex="0"
      type="button"
    >
      Clear Search
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
</DocumentFragment>
`;

exports[`SearchFilterDisplay renders without error 1`] = `
<DocumentFragment>
  <div
    class="css-mqlynq-MuiStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="filterIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.0929 2.57912C1.25675 2.22596 1.61069 2 2.00001 2H22C22.3893 2 22.7433 2.22596 22.9071 2.57912C23.071 2.93229 23.015 3.34845 22.7636 3.64573L15 12.8261V21C15 21.3466 14.8206 21.6684 14.5257 21.8507C14.2309 22.0329 13.8628 22.0494 13.5528 21.8944L9.5528 19.8944C9.21402 19.725 9.00001 19.3788 9.00001 19V12.8261L1.23644 3.64573C0.985046 3.34845 0.929037 2.93229 1.0929 2.57912ZM4.15532 4L10.7636 11.8143C10.9162 11.9948 11 12.2236 11 12.46V18.382L13 19.382V12.46C13 12.2236 13.0838 11.9948 13.2364 11.8143L19.8447 4H4.15532Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </svg>
    <div
      class="css-1d0f7f9-MuiStack-root"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 css-14wykv5-MuiTypography-root"
      >
        Posts containing "Java"
      </div>
    </div>
    <div
      class="MuiBox-root css-i9gxme"
    />
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-qj1k01-MuiButtonBase-root-MuiButton-root"
      data-testid="clear-button"
      tabindex="0"
      type="button"
    >
      Clear Search
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
</DocumentFragment>
`;
